diff --git a/server/middlewares/csp.ts b/server/middlewares/csp.ts
index 369f934..7044ebd 100644
--- a/server/middlewares/csp.ts
+++ b/server/middlewares/csp.ts
@@ -12,6 +12,7 @@ export default function createCSPMiddleware() {
   const defaultSrc: string[] = ["'self'"];
   const scriptSrc: string[] = [];
   const styleSrc: string[] = ["'self'", "'unsafe-inline'"];
+  const baseHost = new URL(env.URL).hostname;
 
   if (env.isCloudHosted) {
     scriptSrc.push("www.googletagmanager.com");
@@ -25,6 +26,8 @@ export default function createCSPMiddleware() {
     scriptSrc.push("localhost:3001");
   } else {
     scriptSrc.push(env.URL);
+    // Allow subdomains to host scripts
+    scriptSrc.push(`https://*.${baseHost}`);
   }
 
   if (env.GOOGLE_ANALYTICS_ID) {
@@ -52,6 +55,9 @@ export default function createCSPMiddleware() {
             ? "'unsafe-inline'"
             : `'nonce-${ctx.state.cspNonce}'`,
         ],
+        // Explicitly add workerSrc, stricter application of standards instead of fallback
+        // to scriptSrc for worker evaluation.
+        workerSrc: [env.URL, `https://*.${baseHost}`],
         mediaSrc: ["*", "data:", "blob:"],
         imgSrc: ["*", "data:", "blob:"],
         frameSrc: ["*", "data:"],
